### DISCUSSION TITLE
I took a backup of S3 with AWS Backup, but "BackupSizeBytes" becomes 0 with the "list-recovery-points-by-resource" command.

### DISCUSSION THREAD

**Discussion by Kobayashi-Riku0226** [aws/aws-cli#8761] [Q&A]
Hello.   

As the title suggests, I took a backup of S3 using AWS Backup.  
I ran the "list-recovery-points-by-backup-vault" command to check the backup size, but "BackupSizeBytes" was 0.  
Is this the right move?  
If it's not correct, why is it 0?  
```
aws backup list-recovery-points-by-resource --resource-arn arn:aws:s3:::s3-bucket-name
```
Below is the response.  
```
{
    "RecoveryPoints": [
        {
            "RecoveryPointArn": "arn:aws:backup:ap-northeast-1:1111111111111:recovery-point:s3-bucket-name-20240622040825-111111111",
            "CreationDate": "2024-06-22T04:08:25.206000+00:00",
            "Status": "COMPLETED",
            "EncryptionKeyArn": "arn:aws:kms:ap-northeast-1:1111111111111:key/yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy",
            "BackupSizeBytes": 0,
            "BackupVaultName": "Default",
            "IsParent": false,
            "ResourceName": "s3-bucket-name",
            "VaultType": "BACKUP_VAULT"
        }
    ]
}
```

**Comment by uldyssian-sh** [comment]
For Amazon S3 backups in AWS Backup, seeing `"BackupSizeBytes": 0` in `list-recovery-points-by-resource` can be normal.

In practice, AWS Backup often does not populate a meaningful “backup size” (and sometimes also reports `bytesTransferred: 0` in events) for S3 recovery points, even when the backup is actually completed and restorable. This has been reported by multiple users for S3 backups. 

Why it can be 0

S3 backups are incremental/object-based, and AWS Backup does not always expose a reliable “size of this recovery point” value via the ListRecoveryPoints APIs for S3. As a result, the field may remain 0 even when the recovery point status is COMPLETED. 

What you can do instead

Confirm the recovery point is usable by checking job/recovery point status (COMPLETED) and performing a test restore of a small prefix.

If you need “how much storage/cost is this backup consuming”, use billing-based reporting (Cost Explorer / CUR) and/or AWS Backup usage/cost reporting rather than `BackupSizeBytes` from ListRecoveryPoints for S3. 

So: your command is fine, and 0 does not automatically mean the backup is empty or failed for S3 — it is generally a limitation/behavior of the reported size fields for S3 backups.

**Reply by uldyssian-sh** [reply]
Thanks! If this helps, please consider marking it as “Helpful” or “Accepted” 😊
